contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -47,448 +43,251 @@ contract FraudProofFacet is StateChannelCommon {
+            return address(0);
+        }
+
+        (bool isTransitionValid, bytes memory encodedModifiedState) = AStateChannelManagerProxy(address(this)).executeStateTransitionOnState(
+            fraudProofVerificationContext.channelId,
+            blockInvalidSTProof.encodedLatestState,
+            fraudBlock.transaction
+        );
+
+        if (isTransitionValid) {
+            return address(0);
+        } 
+
+        if(fraudBlock.stateSnapshotHash == keccak256(encodedModifiedState)){
+            return address(0);
+        }       
+        address signer = StateChannelUtilLibrary.retriveSignerAddress(
+            blockInvalidSTProof.invalidBlock.encodedBlock,
+            blockInvalidSTProof.invalidBlock.signature
+        );
+        return signer;
     }
 
     function _handleBlockOutOfGas(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
         // Implementation for handling BlockOutOfGas proof
         return address(0); // Replace with actual logic
     }
 
-    function _handleTimeoutThreshold(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling TimeoutThreshold proof
-        return address(0); // Replace with actual logic
-    }
+    // ----------------------------------- Timeout Fraud Proofs -----------------------------------
+    function _handleTimeoutThreshold(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal view returns (address) {
+        
+        TimeoutThresholdProof memory timeoutThresholdProof = abi.decode(encodedProof, (TimeoutThresholdProof));
+        BlockConfirmation memory thresholdBlockConfirmation = timeoutThresholdProof.thresholdBlock;
+        Block memory thresholdBlock = abi.decode(thresholdBlockConfirmation.signedBlock.encodedBlock, (Block));
+        Dispute memory originalTimedOutDispute = timeoutThresholdProof.timedOutDispute;
+        
+        bytes32 originalDisputeCommitment = keccak256(abi.encode(
+            originalTimedOutDispute,
+            timeoutThresholdProof.timedOutDisputeTimestamp
+        ));
+
+        (bool isAvailable, bytes32 commitment) = getDisputeCommitment(fraudProofVerificationContext.channelId, originalTimedOutDispute.disputeIndex);
+        if(!isAvailable && commitment != originalDisputeCommitment){
+            return address(0);
+        }
 
-    function _handleTimeoutPriorInvalid(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling TimeoutPriorInvalid proof
-        return address(0); // Replace with actual logic
-    }
+        uint latestStateHeight = _getLatestHeight(originalTimedOutDispute.stateProof);
+        if(latestStateHeight != thresholdBlock.transaction.header.transactionCnt) {
```

Comment:
This is over engineered and not correct - the timedout block is latestStateHeight + 1, but you don't need this at all. 

After checking the dispute commitment, you just need the latestStateSnapshot (the dispute commits to it) so you can provide it as part of the proof - from there you get the participants list and all you have to do is:
check that the block is the correct fork and height as the one registered in Timeout and if so, check the threshold. The UtilityLibrary has a function to check the threshold - for this we don't do virtual voting - so you just provide all the signatures from the BlockConfirmation with the expected participant list given in the latestStateSnapshot